src/main/java/org/opensearch/flowframework/workflow/CreateConnectorStep.java

```diff
@@ -69,8 +80,38 @@ public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) throws I
 
             @Override
             public void onResponse(MLCreateConnectorResponse mlCreateConnectorResponse) {
-                logger.info("Created connector successfully");
-                // TODO Add the response to Global Context
+                try {
+                    logger.info("Created connector successfully");
+                    String workflowId = data.get(0).getWorkflowId();
+                    String workflowStepName = getName();
+                    ResourcesCreated newResource = new ResourcesCreated(workflowStepName, mlCreateConnectorResponse.getConnectorId());
+                    XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
+                    newResource.toXContent(builder, ToXContentObject.EMPTY_PARAMS);
+
+                    // The script to append a new object to the resources_created array
+                    Script script = new Script(
+                        ScriptType.INLINE,
+                        "painless",
+                        "ctx._source.resources_created.add(params.newResource)",
+                        Collections.singletonMap("newResource", newResource)
+                    );
+
+                    flowFrameworkIndicesHandler.updateFlowFrameworkSystemIndexDocWithScript(
+                        WORKFLOW_STATE_INDEX,
+                        workflowId,
+                        script,
+                        ActionListener.wrap(updateResponse -> {
+                            logger.info("updated resources craeted of {}", workflowId);
+                        },
+                            exception -> {
+                                logger.error("Failed to update workflow state with newly created resource: {}", exception.getMessage());
+                            }
+                        )
+                    );
+                } catch (IOException e) {
+                    logger.error("Failed to parse new created resource", e);
+                }
+
```

Comment:
These should be under `future.complete` below.

Reply:
What do you mean by these? do you mean updating the state index? Should it not be a part of the on response here like I have it

Reply:
I meant the whole block should be under `createConnectorFuture.complete` [here](https://github.com/opensearch-project/opensearch-ai-flow-framework/pull/148/files#diff-d523a059b65e2f0888f523cd3329bb573eed90f3826558202b084a1efe2e5856R115)